docs/how-to/iterate-a-spec.rst

```diff
@@ -87,3 +87,30 @@ You can also use this method to only run a subset of the scan:
 >>> len(path)
 0
 
+
+If you need to know whether there is a gap between points
+---------------------------------------------------------
+
+You may need to know where there is a gap between points, so that you can do
+something in the turnaround. For example, if we take the x axis of a grid scan,
+you can see it snakes back and forth:
+
+>>> from scanspec.specs import Line, fly
+>>> grid = fly(Line("y", 0, 1, 2) * ~Line("x", 1, 2, 3), 0.1)
+>>> chunk = grid.path().consume()
+>>> chunk.midpoints["x"]
+array([1. , 1.5, 2. , 2. , 1.5, 1. ])
+
+You can check where the gaps are by using the `Dimension.gap` attribute:
+
+>>> chunk.gap
+array([ True, False, False,  True, False, False])
```

Comment:
I can see the concrete use of this, but in what is a gap in abstract terms? Is it just an arbitrary way of grouping the points? If so, rather than splitting the points into "gapped" and "not gapped" you could split them into groups. For example, in this case:

```python
>>> chunk.midpoints["x"]
array([1. , 1.5, 2. , 2. , 1.5, 1. ])

>>> chunk.groups
array([0, 0, 0, 1, 1, 1])
```
So gapped points are not special. Unsure if this is cleaner or not.

Reply:
I think from a processing point of view the existing structure is more useful